### ISSUE TITLE
PHP8 for NC21/22

### ISSUE THREAD

**Issue by Pilzinsel64** [nextcloud-snap/nextcloud-snap#1654]
Hello.

Would it be an option to update PHP to v8 for the snap Nextcloud 21? I switched to PHP 8 in one of my instances (that I updated to NC21) and the performance improvements are incredible. NC21 itself has a lot of performance improvements but the switch to PHP 8 increaded it significantly more.

**Comment by pheiduck** [comment]
Good idea, but I think we should wait until 8.1 is out or 8.0.1...

**Comment by Pilzinsel64** [comment]
I'm on v8.0.2 aready btw.

**Comment by kyrofa** [comment]
Yeah we can check it out, it's felt pretty sluggish to me lately, too. We've got some other things in flight that should improve performance for v21 as well.

**Comment by kyrofa** [comment]
@Pilzinsel64 any chance this is something you'd like to contribute? I think we're ready for it.

**Comment by pachulo** [comment]
> I switched to PHP 8 in one of my instances (that I updated to NC21) and the performance improvements are incredible.

Out of curiosity, did you set up JIT @Pilzinsel64 ? And if so, which options did you use?

**Comment by Pilzinsel64** [comment]
> @Pilzinsel64 any chance this is something you'd like to contribute? I think we're ready for it.

Oh sorry, I haven't any notifications enabled... But to answer you: My both instances are working well. Had no single issue because of PHP8 since I updated when NC21 has been released.



> Out of curiosity, did you set up JIT @Pilzinsel64 ? And if so, which options did you use?

No, JIT isn't enabled and configured for both of the webservers. I didn't even know that feature. 🤦‍♂️
I could try it out but don't really know what could be the best configuration (reference [here](https://help.creoline.com/doc/php-8-jit-compiler-in-plesk-aktivieren-caUlPGTTJv) because using Plesk). Maybe just wait/ignore until other users share their experience? Even just the switch to PHP8 brought a huge performance boost for me. It's amazing.
